apollo-mockserver-junit5/src/main/java/com/ctrip/framework/apollo/mockserver/MockApolloExtension.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.mockserver;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+import java.io.IOException;
+
+public class MockApolloExtension implements BeforeTestExecutionCallback, AfterTestExecutionCallback, ParameterResolver {
```

Comment:
Is this new module `apollo-mockserver-junit5` necessary? Can we simply put this extension to `apollo-mockserver`?

Reply:
It will be necessary if we want to provide a clean or minimize dependency in the future, but it's not enough because ApolloTestingServer(the prior EmbeddedApollo) depends on legacy okhttp3.mockwebserver, which depends on junit4 and was marked as deprecated now.

I suggest Apollo fix this issue  like newer okhttp mockwebsever did later:
- make old mock server module deprecated.
- a new mockwebserver3 module provide clean MockWebServer with explicit start() and close() method.
- separated  module for mockwebserver3-junit4 and mockwebserver3-junit5

Reply:
You are right. Currently, the apollo-mockserver-junit5 already has the junit4 dependency via okhttp3.mockwebserver, which is why I would suggest we merge these 2 modules, as the junit5 module is not just junit5 :-).
Is it possible we provide a clean apollo-mockserver-junit5 which has no junit4 dependency?

Reply:
Maybe it's should be include in next PR. I  moved all things back to original mock sever module.